pkg/test/parameter_test.go

```diff
@@ -46,10 +48,60 @@ func TestParameterExpandUserFunctions(t *testing.T) {
 	}
 	for _, x := range validExprs {
 		p := NewParam(x[0])
-		res, err := p.Expand(&target.Target{FQDN: x[1], ID: x[2]})
+		res, err := p.Expand(&target.Target{FQDN: x[1], ID: x[2]}, nil)
 		require.NoError(t, err, x[0])
 		require.Equal(t, x[3], res, x[0])
 	}
 	require.NoError(t, UnregisterFunction("CustomFunc"))
 	require.Error(t, UnregisterFunction("NoSuchFunction"))
 }
+
+func TestStepVariablesExpand(t *testing.T) {
+	p := NewParam("{{ StringVar \"step1\" \"string_var\" }}: {{ IntVar \"step1\" \"int_var\" }}")
```

Comment:
i thought the naming we agreed to was with dot separated ids.
```suggestion
	p := NewParam("{{ StringVar \"step1.string_var\" }}: {{ IntVar \"step1.int_var\" }}")
```

i see in the Get() below that they are joined with dot

Reply:
Yes, originally we agreed on dots. I just later thought to give it a try as two parameters.
Probably with dot is better.
Will rework!